### PR TITLE
Add --strict-field-splitting to keep empty field unremoved.

### PR DESCRIPTION
**Pull request by tos-kamiya**
Thank you for the useful tool, pawk.

When I tried to apply pawk to a spreadsheet-like file, I noticed that pawk does not allow empty fields.

E.g, for the following file including an empty field (at line 3):

```
$ cat tbl.csv
model,color,price
A01,blue,100
B03,,200
```

it is difficult to extract values of a column:

```
$ cat tbl.csv | pawk -F, 'f[2]'
price
100
```

So as a (possible) solution, I made an option (--strict-field-splitting or -S for short) to keep empty fields.

```
$ cat tbl.csv | python3 ~/playground/pawk/pawk.py -F, -S 'f[2]'
price
100
200
```

I'm not so fluent in English, so the name of the option (--strict-field-splitting) is perhaps not so good one, sorry.

### CODE REVIEW

**Comment by alecthomas** [comment]
I saw that recently in the code and though to myself that it was a bad idea :\

I think we should just get rid of the current functionality altogether.

I'll bump the major version to indicate a backwards-incompatible change.

**Comment by alecthomas** [comment]
Thanks for the PR! If you want to just make it the default behaviour, I'll merge it.

File: pawk_test.py [hunk 1]
```diff
@@ -75,6 +80,13 @@ def test_integration_multiple_actions():
     assert [r.split()[-1] for r in out.splitlines()] == ['README.md', 'setup.py']
 
 
+def test_integration_csv_empty_fields():
```
Comment (alecthomas) [inline pawk_test.py]: Nice!

File: pawk.py [hunk 2]
```diff
@@ -132,10 +132,7 @@ def _parse_command(arg):
 class Context(dict):
     def apply(self, numz, line, headers=None):
         l = line.rstrip()
-        if self.strict_field_splitting:
-            f = [w for w in l.split(self.delim)]
-        else:
-            f = [w for w in l.split(self.delim) if w]
+        f = [w for w in l.split(self.delim)]
```
Comment (alecthomas) [inline pawk.py]: I think this can just be `f = l.split(self.delim)` now!

Comment (tos-kamiya) [reply pawk.py]: Ah. Of course so!

**Comment by tos-kamiya** [comment]
Fix that issue.
I'm not sure the version naming convention, so I did not change the version (0.8.0).

**Comment by alecthomas** [comment]
Ah, and the flag can be removed too now.

**Comment by alecthomas** [comment]
Thanks!

**Comment by tos-kamiya** [comment]
I'm pleased to hear the patch has been merged.
Thank you!

**Pull request merged**